src/data/finset/basic.lean

```diff
@@ -2085,6 +2152,11 @@ end
 
 @[simp] theorem card_attach {s : finset α} : card (attach s) = card s := multiset.card_attach
 
+@[simp]
+lemma attach_nonempty_iff (s : finset α) :
+  s.attach.nonempty ↔ s.nonempty :=
+by simp [←card_pos]
```

Comment:
```suggestion
by simp [← card_pos]
```

Reply:
You can also prove this more directly using `by simp [finset.nonempty]`, which lets you move this lemma next to the other lemmas about `attach` rather than near the lemmas about `card`, since this isn't about cardinalities. You should also add
```
@[simp] lemma attach_eq_empty_iff (s : finset α) :
  s.attach = ∅ ↔ s = ∅ :=
by simpa [eq_empty_iff_forall_not_mem]
```

Reply:
Hmm... I've PRed quite some code with no space after `←`. Is that a well established convention?

Reply:
My understanding is that the convention is flexible and certainly a few years ago people preferred no space. I guess one should look at the official style guide -- https://leanprover-community.github.io/contribute/style.html -- and actually I can't see this mentioned.